Essentials/src/com/earth2me/essentials/commands/Commandjump.java

```diff
@@ -46,8 +47,7 @@ public void run(final Server server, final User user, final String commandLabel,
 
         final Trade charge = new Trade(this.getName(), ess);
         charge.isAffordableFor(user);
-        user.getTeleport().teleport(loc, charge, TeleportCause.COMMAND);
-        throw new NoChargeException();
+        user.getAsyncTeleport().teleport(loc, charge, TeleportCause.COMMAND, getNewExceptionFuture(user.getSource(), commandLabel), new CompletableFuture<>());
```

Comment:
As far as I can tell, the `throw new NoChargeException()` is to prevent players paying the command cost for both the `/jump` command and the teleport itself. Does this change alter this behaviour (or am I reading it wrong)?